src/App.tsx

```diff
@@ -1,7 +1,35 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type State = {
+  key: string,
+};
+// eslint-disable-next-line react/prefer-stateless-function
+export class App extends React.Component {
+  state: State = {
+    key: '',
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', this.handleKeyPress);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', this.handleKeyPress);
+  }
+
+  handleKeyPress = (event: KeyboardEvent) => {
+    this.setState({ key: event.key });
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <p className="App__message">
+          {this.state.key
```

Comment:
Use a destructuring for `key`

Reply:
i fixed key in second commit, but on demo is still completed task of 1st commit. And why I need to desctructing key if its the only variable of state in return jsx?

Reply:
And why I need to desctructing key if its the only variable of state in return jsx?


It makes the code a bit more readable. If there are a lot of values in the state it's better to destructure all values and not repeat `this.state` everywhere